xwiki-platform-core/xwiki-platform-csrf/src/main/java/org/xwiki/csrf/internal/DefaultCSRFToken.java

```diff
@@ -193,9 +197,14 @@ public String getResubmissionURL()
 
         // back URL is the URL of the document that was about to be modified, so in most
         // cases we can redirect back to the correct document (if the user clicks "no")
-        String backUrl = getDocumentURL(this.docBridge.getCurrentDocumentReference(), null);
+        DocumentReference currentDocRef = this.docBridge.getCurrentDocumentReference();
+        // If the current document is null (on the login page for example), let's redirect to the default page
+        if (currentDocRef == null) {
+            currentDocRef = resolver.resolve("");
```

Comment:
Note 1: You can always resolve any reference, including null so there's no need for an IF. We should mention it in the javadoc.
Note 2: getDocumentURL() should work with null already. I think there's a bug in `DAB#getDocumentURL` which leads to an infinite loop if `this.getContext().getDoc().getDocumentReference()` is null. We need to fix that IMO but I'm not sure what behavior we should have then. Maybe do as you did and go to the default page.

@tmortagne WDYT?

Reply:
Actually, I can't pass `currentDocRef` to the resolver because it can only accept strings and even if the DocumentReference can be converted to string, Java doesn't do it implicitly. By the way, I can only pass null or empty string to get the current behaviour.